### PR TITLE
Feature/task11 retrieve information for general wrestler table

### PR DESCRIPTION
**Pull request by TiagoVidal91**
- Creation of endpoint for retuning a list of wrestlers, filtered by lockerroom, brand, number of victories, losses and streaks, and percentage of victories.
- Changes made to the process of fetching data from database (JPA to JDBI)
- Unit Tests
- Made changes to the get JWT method to return null instead of an error, since when the user did not login he would get an error when he wasn't supposed to.
- Changes the Units Tests to reflect this

### CODE REVIEW

File: src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Services/WrestlerServiceImpl.java [hunk 1]
```diff
@@ -1,106 +1,62 @@
...
-        modelMapper.getConfiguration().setSkipNullEnabled(true);
-        modelMapper.getConfiguration().setPreferNestedProperties(false);
-        defineWrestlerMapper(propertyMapper);
+        this.validationUtils = validationUtils;
     }
 
     @Override
     public Wrestler addNewWrestler(Wrestler wrestler) {
-        verifyIfWrestlerExists(wrestler);
+        //verifyIfWrestlerExists(wrestler);
```
Comment (clara-a-correia) [inline src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Services/WrestlerServiceImpl.java:30]: add TODO to be able to find - maybe even associate an issue/task number if already existing

File: src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Repository/WrestlerRepoImplTest.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+    private int page = 1;
+    private int size = 10;
+    private Integer brandId = 1;
+    private Integer lockerId = 1;
+    private String orderBy = "in_ring_name";
+    private String orderDirection = "ASC";
+
+    @Test
+    public void findWrestlerListByRingNameASC() {
```
Comment (clara-a-correia) [inline src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Repository/WrestlerRepoImplTest.java]: Simplify methods according to the variable being changed. 
p.ex you can combine findWrestlerListByRingNameASC and DESC, etc 
keep the complexity you find necessary for it to be readable and expandable

**Comment by clara-a-correia** [comment]
Consider changing the methodology of the unit tests for them to act on mock data rather than on production tables. A change in the tables could affect the test results, and your intent should be to test the logic of the code and not the results themselves

**Review by clara-a-correia — CHANGES_REQUESTED** [review]

**Comment by TiagoVidal91** [comment]
- Made changes to the main query, since it did not return the correct results for the total of victories, total losses and percentage.
- Implemented H2 to recriate the database in memory for testing porpuses (Added the data and schema sql for this porpuse)
- Made changes to the Unit tests to be less verbose and more efficient
- Included more Unit tests that were not considered then (order by total victories, total losses, percentage and streak)
- Added a new validation to the findAllWrestlersByParams method on the Service class to verify if the user is sending the correct param to be converted to the table name

File: src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Repository/WrestlerRepoImplTest.java (hunk 2 above)
Comment (TiagoVidal91) [reply src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Repository/WrestlerRepoImplTest.java]: Done

File: src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Services/WrestlerServiceImpl.java (hunk 1 above)
Comment (TiagoVidal91) [reply src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Services/WrestlerServiceImpl.java:30]: Done!

**Review by clara-a-correia — APPROVED** [review]

**Pull request merged**